Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Profile.hs

```diff
@@ -176,15 +178,19 @@ getPersonDetails (personId, _) mbToss = do
   frntndfgs <- if useCACConfig then getFrontendConfigs person mbToss else return $ Just DAKM.empty
   let mbMd5Digest = T.pack . show . MD5.md5 . DA.encode <$> frntndfgs
   isSafetyCenterDisabled_ <- SLP.checkSafetyCenterDisabled person
-  return $ makeProfileRes decPerson tag mbMd5Digest isSafetyCenterDisabled_
+  newCustomerReferralCode <- DR.generateCustomerReferralCode
+  when (isNothing person.customerReferralCode) $ do
+    void $ QPerson.updateCustomerReferralCode personId newCustomerReferralCode
+  return $ makeProfileRes decPerson tag mbMd5Digest isSafetyCenterDisabled_ newCustomerReferralCode
   where
-    makeProfileRes Person.Person {..} disability md5DigestHash isSafetyCenterDisabled_ =
+    makeProfileRes Person.Person {..} disability md5DigestHash isSafetyCenterDisabled_ newCustomerReferralCode =
```

Comment:
send existing or generated whatever is the case from above to this fn and directly assign value to `customerReferralCode` instead of `customerReferralCode <|> (Just newCustomerReferralCode)`